### PR TITLE
Add C++ accelerometer example and use LinearFilter

### LINKED ISSUE #107: C++ example in Accelerometers - Software
See the digital linear filters snippet.
https://frc-docs.readthedocs.io/en/develop/docs/software/sensors/accelerometers-software.html#using-accelerometers-in-code

### PR DESCRIPTION
**Pull request by prateekma**
Fixes #107 
Also replaced deprecated `LinearDigitalFilter` with new `LinearFilter`. 
The link to the docs doesn't have the new `LinearFilter` yet, so I left the link as is.

### CODE REVIEW

File: source/docs/software/sensors/accelerometers-software.rst [hunk 1]
```diff
@@ -210,7 +210,7 @@ For detecting collisions, it is often more robust to measure the jerk than the a
             prevYAccel = yAccel;
         }
 
-Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearDigitalFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:
+Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:
```
Comment (AustinShalit) [inline source/docs/software/sensors/accelerometers-software.rst]: Note should be added to update the link to the java doc

Comment (prateekma) [reply source/docs/software/sensors/accelerometers-software.rst]: Note added in 770de7e

File: source/docs/software/sensors/accelerometers-software.rst [hunk 2]
```diff
@@ -220,30 +220,22 @@ Most accelerometers legal for FRC use are quite noisy, and it is often a good id
...
-            double filteredXAccel = xAccelFilter.pidGet();
+            double filteredXAccel = xAccelFilter.calculate(accelerometer.getX());
         }
 
     .. code-tab:: c++
 
-        .. todo:: C++ example
+        frc::BuiltInAccelerometer accelerometer;
+
+        // Create a LinearDigitalFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop
```
Comment (AustinShalit) [inline source/docs/software/sensors/accelerometers-software.rst]:
```suggestion
        // Create a LinearFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop
```

Comment (prateekma) [reply source/docs/software/sensors/accelerometers-software.rst]: Whoops, my bad for missing this. Fixed in 471f8ce

Comment (prateekma) [reply source/docs/software/sensors/accelerometers-software.rst]: My bad for missing this, fixed in 471f8ce

**Comment by prateekma** [comment]
@AustinShalit Do I need to make any more changes..?

File: source/docs/software/sensors/accelerometers-software.rst [hunk 3]
```diff
@@ -210,40 +210,34 @@ For detecting collisions, it is often more robust to measure the jerk than the a
...
+            double filteredXAccel = xAccelFilter.calculate(accelerometer.getX());
         }
 
     .. code-tab:: c++
 
-        .. todo:: C++ example
+        frc::BuiltInAccelerometer accelerometer;
+
+        // Create a LinearFilter that will calculate a moving average of the measured X acceleration over the past 10 iterations of the main loop
+        auto filter = frc::LinearFilter::MovingAverage(10);
```
Comment (calcmogul) [inline source/docs/software/sensors/accelerometers-software.rst]:
```suggestion
        auto xAccelFilter = frc::LinearFilter::MovingAverage(10);
```

File: source/docs/software/sensors/accelerometers-software.rst [hunk 4]
```diff
@@ -210,40 +210,34 @@ For detecting collisions, it is often more robust to measure the jerk than the a
             prevYAccel = yAccel;
         }
 
-Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearDigitalFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:
+Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/filters/LinearDigitalFilter.html>`__) to reduce the noise:
```
Comment (calcmogul) [inline source/docs/software/sensors/accelerometers-software.rst]:
```suggestion
Most accelerometers legal for FRC use are quite noisy, and it is often a good idea to combine them with the :code:`LinearFilter` class (`Java <https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/LinearFilter.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/release/docs/cpp/classfrc_1_1LinearFilter.html>`__) to reduce the noise:
```

Comment (Daltz333) [reply source/docs/software/sensors/accelerometers-software.rst]: It says those links are invalid

Comment (calcmogul) [reply source/docs/software/sensors/accelerometers-software.rst]: Those are the correct links on wpilib master, at least.

**Comment by prateekma** [comment]
I removed the `TODO` for updating the URL since we already have the updated URL in place.

**Pull request merged**